src/servers/middlewares/kratos-router.ts

```diff
@@ -163,4 +164,40 @@ kratosRouter.post(
   }),
 )
 
+kratosRouter.post(
+  "/subjectToUserId",
+  wrapAsyncToRunInSpan({
+    namespace: "subjectToUserId",
+    fn: async (req: express.Request, res: express.Response) => {
+      // TODO check basic auth
+      // console.log("missing authorization header")
+      // res.status(401).send("missing authorization header")
+      // return
+
+      let userId: UserId | undefined
+      try {
+        const { subject } = req.body
+        const maybeUserId = checkedToUserId(subject)
+        const maybeDeviceId = checkedToDeviceId(subject)
+        if (!(maybeUserId instanceof ValidationError)) {
+          userId = maybeUserId
+        } else if (!(maybeDeviceId instanceof ValidationError)) {
+          const deviceId = maybeDeviceId
+          const deviceUser = await UsersRepository().findByDeviceId(deviceId)
+          if (!(deviceUser instanceof Error)) userId = deviceUser.id
+        }
+        res.status(200).json({
+          subject, // subject: userId, // <=== this breaks if you try to mutate the subject (i.e changing deviceId subject to userId)
+          extra: {
+            userId, // <==== need to use extra
+          },
```

Comment:
@nicolasburtey ,

you cannot mutate the subject (a JSON validation error is thrown by oathkeeper) so I added an extra field called userId. 

Ory added an AI bot in their docs! The bot helped answer the question why changing the subject caused errors. (some keys, like subject cannot be changes)
<img width="568" alt="Screen Shot 2023-05-16 at 8 43 30 PM" src="https://github.com/GaloyMoney/galoy/assets/1273575/fdce654a-6306-40f6-b081-b6b28702f342">

Reply:
then we can do a `userId = token.userId ?? token.sub` I guess. not ideal but probably better flow that what we had previously.